src/Integration/VsSessionHost.cs

```diff
@@ -0,0 +1,272 @@
+                throw new ArgumentNullException(nameof(sonarQubeService));
+            }
+
+            if (solutionTacker == null)
+            {
+                throw new ArgumentNullException(nameof(solutionTacker));
+            }
+
+            if (uiDispatcher == null)
+            {
+                throw new ArgumentNullException(nameof(uiDispatcher));
+            }
+
+            if (solutionBinding == null)
+            {
+                throw new ArgumentNullException(nameof(solutionBinding));
+            }
+
+            this.serviceProvider = serviceProvider;
+            this.VisualStateManager = state ?? new StateManager(this, new TransferableVisualState());
+            this.progressStepRunner = progressStepRunner ?? this;
+            this.UIDIspatcher = uiDispatcher;
+            this.SonarQubeService = sonarQubeService;
+            this.solutionBinding = solutionBinding;
+            this.solutionTacker = solutionTacker;
+            this.solutionTacker.ActiveSolutionChanged += this.OnActiveSolutionChanged;
+        }
+
+        #region IProgressStepRunnerWrapper
+        void IProgressStepRunnerWrapper.AbortAll()
+        {
+            ProgressStepRunner.AbortAll();
+        }
+
+        void IProgressStepRunnerWrapper.ChangeHost(IProgressControlHost host)
+        {
+            ProgressStepRunner.ChangeHost(host);
+        }
+        #endregion
+
+        #region IHost
+        public Dispatcher UIDIspatcher { get; }
+
+        public IStateManager VisualStateManager { get;  }
+
+        public ISonarQubeServiceWrapper SonarQubeService { get; }
+
+        public IConnectSection ActiveSection { get; private set; }
+
+        public void SetActiveSection(IConnectSection section)
+        {
+            if (section == null)
+            {
+                throw new ArgumentNullException(nameof(section));
+            }
+
+            Debug.Assert(this.ActiveSection == null, "Already attached. Detach first");
+            this.ActiveSection = section;
+            this.VisualStateManager.SyncCommandFromActiveSection();
+
+            this.TransferState();
+
+            if (this.resetBindingWhenAttaching)
+            {
+                this.resetBindingWhenAttaching = false;
+
+                // The connect section activated after the solution is opened, 
+                // so reset the binding if applicable. No reason to abort since
+                // this is the first time after the solution was opened so that 
+                // we switched to the connect section.
+                this.ResetBinding(abortCurrentlyRunningWorklows: false);
+            }
+        }
+
+        public void ClearActiveSection()
+        {
+            if (this.ActiveSection == null) // Can be called multiple times
+            {
+                return;
+            }
+
+            this.ActiveSection.ViewModel.State = null;
+            this.ActiveSection = null;
+            this.VisualStateManager.SyncCommandFromActiveSection();
```

Comment:
Should this call to `SyncCommandFromActiveSection` happen here?
When we navigate away from the section in the Team Explorer, the SectionController is disposed, which in turn calls this method. `SyncCommandFromActiveSection` then actually goes off to create new context menu commands (which aren't accessible until we re-attach an active section).